pkg/policy/selectorcache_test.go

```diff
@@ -510,17 +524,29 @@ func newDummyIdentityNotifier() *dummyIdentityNotifier {
 	}
 }
 
-// RegisterForIdentityUpdates starts managing this selector.
-func (d *dummyIdentityNotifier) RegisterForIdentityUpdates(selector api.FQDNSelector) (identities []identity.NumericIdentity) {
+// Lock must be held during any calls to RegisterForIdentityUpdatesLocked or
+// UnregisterForIdentityUpdatesLocked.
+func (d *dummyIdentityNotifier) Lock() {
+	d.mutex.Lock()
+}
+
+// Unlock must be called after calls to RegisterForIdentityUpdatesLocked or
+// UnregisterForIdentityUpdatesLocked are done.
+func (d *dummyIdentityNotifier) Unlock() {
+	d.mutex.Unlock()
+}
+
+// RegisterForIdentityUpdatesLocked starts managing this selector.
+func (d *dummyIdentityNotifier) RegisterForIdentityUpdatesLocked(selector api.FQDNSelector) (identities []identity.NumericIdentity) {
```

Comment:
nit for future follow-up: Seems like this and `pkg/k8s/network_policy_test.go ` have dummyIdentityNotifier implementations, we could consider refactoring them into `pkg/testutils` to deduplicate.

Reply:
Done.